site/src/pages/WorkspacesPage/WorkspacesPage.tsx

```diff
@@ -21,6 +25,37 @@ const WorkspacesPage: FC = () => {
     ...pagination,
     query: filterProps.filter.query,
   })
+
+  const experimentEnabled = useIsWorkspaceActionsEnabled()
+  // If workspace actions are enabled we need to fetch the locked
+  // workspaces as well. This lets us determine whether we should
+  // show a banner to the user indicating that some of their workspaces
+  // are at risk of being deleted.
+  useEffect(() => {
+    if (experimentEnabled) {
+      const includesLocked = filterProps.filter.query.includes("locked_at")
+      const lockedQuery = includesLocked
+        ? filterProps.filter.query
+        : filterProps.filter.query + " locked_at:1970-01-01"
+
+      if (includesLocked && data) {
+        setLockedWorkspaces(data.workspaces)
+      } else {
+        getWorkspaces({ q: lockedQuery })
+          .then((resp) => {
+            setLockedWorkspaces(resp.workspaces)
+          })
+          .catch(() => {
+            // TODO
+          })
+      }
+    } else {
+      // If the experiment isn't included then we'll pretend
+      // like locked workspaces don't exist.
+      setLockedWorkspaces([])
+    }
+  }, [experimentEnabled, data, filterProps.filter.query])
+
```

Comment:
This logic feels very similar to the `useWorkspacesToBeLocked`, why not use it here as well?

Reply:
The `useWorkspacesToBeLocked` takes a template which the workspaces page doesn't mandate. I could refactor it, but I'd rather merge this as-is and go back and clean up things as we move it out of experimental.